### PR TITLE
Shamir's Secret Sharing cryptographic algorithm in C++

### PR DESCRIPTION
**Pull request by pganguli**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have.

### Description

Adds Shamir's Secret Sharing cryptographic algorithm in C++.

### Checklist

- [X] I've read the contribution guidelines.
- [X] I've checked the issue list before deciding what to submit.
- [X] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue #2140 .

### CODE REVIEW

**Comment by HarshCasper** [comment]
Thanks for making your Pull Request!

PRs would be reviewed after the start of the Coding Period. Thanks!

**Comment by pganguli** [comment]
> Thanks for making your Pull Request!
> 
> PRs would be reviewed after the start of the Coding Period. Thanks!

As per the [GSSOC'21 schedule](https://gssoc.girlscript.tech/schedule.html), I was under the impression that the coding period would have started from the 1st of March. Can you please let me know when it really starts from? Thanks!

**Comment by HarshCasper** [comment]
> > Thanks for making your Pull Request!
> > PRs would be reviewed after the start of the Coding Period. Thanks!
> 
> As per the [GSSOC'21 schedule](https://gssoc.girlscript.tech/schedule.html), I was under the impression that the coding period would have started from the 1st of March. Can you please let me know when it really starts from? Thanks!

The timeline has been updated and it would start from March 8, 2021. Keep an eye on the Mail.

File: C-Plus-Plus/cryptography/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+[Shamir's Secret Sharing algorithm](cryptography/Shamir_Secret_Sharing.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/cryptography/README.md]: We don't add programs on the individual sub-directory READMEs as of now. Please make the README changes on the language-specific README. Have a look at the existing PRs to understand this.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Great Work! Ready to merge :tada: 

One tiny change requested, and it would be merged!

File: C-Plus-Plus/cryptography/README.md (hunk 1 above)
Comment (pganguli) [reply C-Plus-Plus/cryptography/README.md]: Got it, thanks. Resolved now.

**Pull request merged**